### ISSUE TITLE
Support for distributed training

### ISSUE THREAD

**Issue by Ending2015a** [netw0rkf10w/CRF#7]
Hello, thank you for this amazing work!

I want to use this CRF package for distributed training. I wonder if it is OK to use it to train a segmentation model across many GPUs or processes. For example, does it contain any operations like batch normalization that needs to perform computations across batches?

**Comment by netw0rkf10w** [comment]
Hi @Ending2015a. Thanks for your interest!
You can safely use `CRF` for any kind of distributed training (e.g., Pytorch DDP or DataParallel). I will try to release my semantic segmentation code soon. The code is actually ready but it takes a lot of time to write a good README, which I started but had to stop to work on some deadlines.
Please feel free to ask if you have any questions.

**Comment by Ending2015a** [comment]
Thank you!
